alpha/veneer/composite/builder.go

```diff
@@ -0,0 +1,317 @@
+}
+
+var _ Builder = &RawBuilder{}
+
+func NewRawBuilder(builderCfg BuilderConfig) *RawBuilder {
+	return &RawBuilder{
+		builderCfg: builderCfg,
+	}
+}
+
+func (rb *RawBuilder) Build(dir string, vd VeneerDefinition) error {
+	if vd.Schema != RawVeneerBuilderSchema {
+		return fmt.Errorf("schema %q does not match the raw veneer builder schema %q", vd.Schema, RawVeneerBuilderSchema)
+	}
+	// Parse out the raw veneer configuration
+	rawConfig := &RawVeneerConfig{}
+	err := json.Unmarshal(vd.Config, rawConfig)
+	if err != nil {
+		return fmt.Errorf("unmarshalling raw veneer config: %w", err)
+	}
+
+	// validate the raw config fields
+	valid := true
+	validationErrs := []string{}
+	if rawConfig.Input == "" {
+		valid = false
+		validationErrs = append(validationErrs, "raw veneer config must have a non-empty input (veneerDefinition.config.input)")
+	}
+
+	if rawConfig.Output == "" {
+		valid = false
+		validationErrs = append(validationErrs, "raw veneer config must have a non-empty output (veneerDefinition.config.output)")
+	}
+
+	if !valid {
+		return fmt.Errorf("raw veneer configuration is invalid: %s", strings.Join(validationErrs, ","))
+	}
+
+	// build the container command
+	containerCmd := exec.Command(rb.builderCfg.ContainerCfg.ContainerTool,
+		"run",
+		"--rm",
+		"-v",
+		fmt.Sprintf("%s:%s", rb.builderCfg.CurrentDirectory, rb.builderCfg.ContainerCfg.WorkingDir),
+		"--entrypoint=cat", // This assumes that the `cat` command is available in the container -- Should we also build a `... render-veneer raw` command to ensure consistent operation? Does OPM already have a way to render a raw FBC?
+		rb.builderCfg.ContainerCfg.BaseImage,
+		path.Join(rb.builderCfg.ContainerCfg.WorkingDir, rawConfig.Input))
+
+	return build(containerCmd, path.Join(dir, rawConfig.Output), rb.builderCfg.OutputType)
+}
+
+func (rb *RawBuilder) Validate(dir string) error {
+	return validate(rb.builderCfg.ContainerCfg, path.Join(rb.builderCfg.CurrentDirectory, dir))
+}
+
+type CustomBuilder struct {
+	builderCfg BuilderConfig
+}
+
+var _ Builder = &CustomBuilder{}
+
+func NewCustomBuilder(builderCfg BuilderConfig) *CustomBuilder {
+	return &CustomBuilder{
+		builderCfg: builderCfg,
+	}
+}
+
+func (cb *CustomBuilder) Build(dir string, vd VeneerDefinition) error {
+	if vd.Schema != CustomVeneerBuilderSchema {
+		return fmt.Errorf("schema %q does not match the custom veneer builder schema %q", vd.Schema, CustomVeneerBuilderSchema)
+	}
+	// Parse out the raw veneer configuration
+	customConfig := &CustomVeneerConfig{}
+	err := json.Unmarshal(vd.Config, customConfig)
+	if err != nil {
+		return fmt.Errorf("unmarshalling custom veneer config: %w", err)
+	}
+
+	// validate the custom config fields
+	if customConfig.Command == "" {
+		return errors.New("custom veneer configuration is invalid: custom veneer config must have a non-empty command (veneerDefinition.config.command)")
+	}
+
+	// build the command to execute
+	// TODO: should the command be run within the container?
```

Comment:
The intent here was just to provide an 'escape hatch' for custom processing.  I think it's totally safe for now to assume that it runs locally.

Reply:
Sgtm, removed TODO comment in commit https://github.com/operator-framework/operator-registry/pull/1054/commits/bfde6b3074cc7e7a083048f085c2aac7e22c4c70